sympy/functions/combinatorial/factorials.py

```diff
@@ -890,17 +898,17 @@ def _eval(self, n, k):
                     result = result * d // i
                 return Integer(result)
             else:
-                d, result = n - k, 1
-                for i in range(1, k + 1):
-                    d += 1
-                    result *= d
-                    result /= i
-                return result
+                return Mul(*[n + i for i in range(1 - k, 1)])/factorial(k)
 
     @classmethod
     def eval(cls, n, k):
         n, k = map(sympify, (n, k))
+        kk = next(ordered((k, n - k)))
+        if kk.is_negative == k.is_negative:
```

Comment:
perhaps we should make this
n, k = map(sympify, (n, k))
if k.is_negative and k.is_integer:
    return S.Zero  # binomial(n, k) == binomial(n - 1, k) + binomial(n - 1, k - 1)
if n == k and n.is_real and None not in (k.is_negative, k.is_integer):
    return S.One  # binomial(n, k) == binomial(n, n - k)
k = next(ordered((k, n - k)))
```
but then we have to be sure that  `is_integer` does not answer unless the real, integer and sign status of k or n - k is known. (binomial(oo,oo) is not 1 so binomial(x,x) != 1 unless we know that x is real, not extended_real.)

Reply:
Yeahh!, Okay.